### PR TITLE
adding upload download option for chat client config 

### PR DESCRIPTION
**Pull request by hiteshghuge**

### CODE REVIEW

File: tests/unit_test/utility_test.py [hunk 1]
```diff
@@ -176,21 +179,30 @@ async def test_upload_and_save(self):
...
             )
-        )
-        training_data = processor.load_nlu("test_load_from_path_yml_training_files")
-        story_graph = processor.load_stories("test_load_from_path_yml_training_files")
-        domain = processor.load_domain("test_load_from_path_yml_training_files")
-        config = processor.load_config("test_load_from_path_yml_training_files")
-        http_action = processor.load_http_action("test_load_from_path_yml_training_files")
-        training_data_path = Utility.write_training_data(training_data, domain, config, story_graph, None, http_action)
-        assert os.path.exists(training_data_path)
+            training_data = processor.load_nlu("test_load_from_path_yml_training_files")
```
Comment (udit-pandey) [inline tests/unit_test/utility_test.py:199]: move out of with block

**Review by udit-pandey — CHANGES_REQUESTED** [review]
Tests:
1. Upload without chat client config
2. Upload with hat client config
3. upload only chat client config
4. validate chat client validation erros are being logged
5.  upload only chat client config should not trigger event
6. upload empty chat client file, empty dict chat client file
7. upload chat client config in append mode

File: tests/unit_test/utility_test.py (hunk 1 above)
Comment (udit-pandey) [reply tests/unit_test/utility_test.py:199]: rebase from digiteinfotech/master

File: kairon/shared/data/processor.py [hunk 2]
```diff
@@ -3128,14 +3138,21 @@ def save_data_without_event(self, data_home_dir: Text, bot: Text, user: Text, ov
             config = Utility.read_yaml(config_path)
             errors = TrainingDataValidator.validate_rasa_config(config)
             error_summary['config'] = errors
+        if os.path.exists(chat_client_config_path):
+            chat_config = Utility.read_yaml(chat_client_config_path)
+            errors = TrainingDataValidator.validate_chat_client_config(chat_config)
+            error_summary['chat_client_config'] = errors
 
         if not validation_failed and not error_summary.get('config'):
```
[the remark this answers is not in the document]
Comment (hiteshghuge) [reply kairon/shared/data/processor.py:3146]: added

File: template/use-cases/Covid-19/chat_client_config.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
+name: kairon
```
[the remark this answers is not in the document]
Comment (hiteshghuge) [reply template/use-cases/Covid-19/chat_client_config.yml:1]: Not possible

File: kairon/shared/data/processor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hiteshghuge) [reply kairon/shared/data/processor.py]: done

File: tests/unit_test/data_processor/data_processor_test.py [hunk 4]
```diff
@@ -2065,69 +2098,82 @@ def test_delete_rules_no_rules(self):
...
+            config = UploadFile(filename="config.yml", file=BytesIO(config_content))
+            domain = UploadFile(filename="domain.yml", file=BytesIO(domain_content))
+            chat_client_config = UploadFile(filename="chat_client_config.yml", file=BytesIO(chat_client_config_content))
+            await processor.upload_and_save(nlu, domain, stories, config, None, None, chat_client_config, "test_upload_and_save",
+                                            "rules_creator")
+            assert len(list(Intents.objects(bot="test_upload_and_save", user="rules_creator"))) == 6
+            assert len(list(Stories.objects(bot="test_upload_and_save", user="rules_creator"))) == 1
+            assert len(list(Responses.objects(bot="test_upload_and_save", user="rules_creator"))) == 3
+            assert len(
+                list(TrainingExamples.objects(intent="greet", bot="test_upload_and_save", user="rules_creator"))) == 2
```
[the remark this answers is not in the document]
Comment (hiteshghuge) [reply tests/unit_test/data_processor/data_processor_test.py]: done

**Pull request closed without merging**